src/mediapipe_internal/mediapipegraphdefinition.cpp

```diff
@@ -148,10 +179,54 @@ Status MediapipeGraphDefinition::create(std::shared_ptr<MediapipeGraphExecutor>&
         return status;
     }
     SPDLOG_DEBUG("Creating Mediapipe graph executor: {}", getName());
-    pipeline = std::make_shared<MediapipeGraphExecutor>(getName(), std::to_string(getVersion()), this->config);
+
+    bool passKfsRequestFlag = false;
+    // Check if we are passing whole KFS request and response
+    status = this->setKFSPassthrough(passKfsRequestFlag);
+    if (!status.ok()) {
+        SPDLOG_DEBUG("Failed to execute mediapipe graph: {} KFS passthrough mode is misconfigured.", getName());
+        return status;
+    }
+
+    pipeline = std::make_shared<MediapipeGraphExecutor>(getName(), std::to_string(getVersion()),
+        this->config, passKfsRequestFlag, this->inputNames, this->outputNames);
     return status;
 }
 
+Status MediapipeGraphDefinition::setKFSPassthrough(bool& passKfsRequestFlag) {
+    passKfsRequestFlag = false;
+    if (this->config.input_stream().size() == 1) {
+        std::string firstName = this->config.input_stream()[0];
+        if (startsWith(firstName.c_str(), "REQUEST:")) {
```

Comment:
What will happen if we will have

input_stream: in
output_stream: RESPONSE:out

and is it consistent with

input_stream: REQUEST:in
output_stream: out

Reply:
Fixed.